src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -277,8 +281,16 @@ typedef struct DistributedExecution
 	 */
 	AttInMetadata *attributeInputMetadata;
 	char **columnArray;
+
+	/*
+	 * isOutsideTransaction is used to determine if we should begin a transaction
+	 * or not. For example, we do not begin a transaction for repartition specific
+	 * transactions.
+	 */
+	bool isOutsideTransaction;
```

Comment:
After some discussion with Onder, we agreed that we effectively have 3 situations:
- We need tasks to be executed in transaction blocks (isTransaction is true)
- We need tasks to be executed outside transaction blocks (isOutsideTransaction is true)
- We don't care whether or not tasks are executed in transaction blocks (both false)

It would make sense for these 3 situations to be captured in a single enum, rather than 2 bools.

For instance `RemoteTransactionBlocksUsage useRemoteTransactionBlocks` with `REMOTE_TRANSACTION_BLOCKS_REQUIRED`, `REMOTE_TRANSACTION_BLOCKS_ALLOWED`, `REMOTE_TRANSACTION_BLOCKS_DISALLOWED`.